vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,265 @@
+---
+title: "Projecting infectious disease incidence using early outbreak data"
+author: "James Azam, Sebastian Funk"
+output:
+  bookdown::html_vignette2:
+    fig_caption: yes
+    code_folding: show
+pkgdown:
+  as_is: true
+bibliography: references.bib
+link-citations: true
+vignette: >
+  %\VignetteIndexEntry{Projecting infectious disease incidence using early outbreak data}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, 
+                      message = FALSE, 
+                      warning = FALSE, 
+                      collapse = TRUE,
+                      comment = "#>",
+                      dpi = 300
+                      )
+
+```
+
+## Overview
+
+Branching processes can be used to project infectious disease trends provided 
+we can characterize the distribution of times between 
+successive cases (serial interval), and the distribution of secondary cases 
+produced by a single individual (offspring distribution). Such simulations can be achieved in `bpmodels` with the `chain_sim()` function. @Pearson2020, and 
+@abbott2020 illustrate its application to COVID-19. 
+
+The purpose of this vignette is to use early data on COVID-19 in South Africa [@marivate2020] to illustrate how `bpmodels` can be used to forecast an outbreak. 
+
+
+Let's load the required packages
+
+```{r packages, include=TRUE}
+library("bpmodels")
+library('dplyr')
+library('ggplot2')
+library('lubridate')
+```
+
+### The data
+
+We will get and clean the first $15$ days of the COVID-19 
+outbreak in South Africa to seed the simulation for this example.
+
+```{r data, message=FALSE}
+data_url <- 'https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_timeline_confirmed.csv'
+
+#Read the data in using the url
+covid19_sa <- read.csv(data_url)
+
+# Subset the first 15 days and count the number of cases per date
+covid19_sa <- covid19_sa %>%
+  dplyr::select(date) %>%
+  dplyr::mutate(date = lubridate::dmy(date)) %>%
+  dplyr::filter(date <= lubridate::dmy("20-03-2020")) %>%
```

Comment:
Suggest to not hardcode this, rather use `date <= min(date) + lubridate::days(15)`. Also no need for explicit namespacing for {dplyr} ({lubridate} namespaced to clarify where the `days()` function is from).

Reply:
Thanks for this.